install/error-handling.sh

```diff
@@ -15,9 +41,30 @@ cleanup () {
     return 0;
   fi
   DID_CLEAN_UP=1
-
+  local retcode=$?
+  local cmd="${BASH_COMMAND}"
   if [[ "$1" != "EXIT" ]]; then
-    echo "An error occurred, caught SIG$1 on line $2";
+    set +o xtrace
+    printf -v err '%s' "Error in ${BASH_SOURCE[1]}:${BASH_LINENO[1]}."
+    printf -v cmd_exit '%s' "'$cmd' exited with status $retcode"
+    printf '%s\n%s\n' "$err" "$cmd_exit"
+    local stack_depth=${#FUNCNAME[@]}
+    local traceback=""
+    if [ $stack_depth -gt 2 ]; then
+      for ((i=$(($stack_depth - 1)),j=1;i>0;i--,j++)); do
+          local indent="$(yes a | head -$j | tr -d '\n')"
+          local src=${BASH_SOURCE[$i]}
+          local lineno=${BASH_LINENO[$i-1]}
+          local funcname=${FUNCNAME[$i]}
+          printf -v traceback '%s\n' "$traceback${indent//a/-}>$src:$funcname:$lineno"
+      done
+    fi
+    echo "$traceback"
+
+    if [ "$REPORT_ERRORS" == 1 ]; then
+      local traceback_hash=$(echo -n $traceback | docker run --rm busybox md5sum | cut -d' ' -f1)
```

Comment:
Will this hash the same if I hit the error from an installation at `/home/sentry` and a second at `/usr/local/sentry`?

Reply:
It will, I confirmed this is just the file name not the full path.